packages/lex_service/src/lexicon/resolvers/lexicon.resolver.ts

```diff
@@ -16,6 +17,20 @@ export class LexiconResolver {
     return this.lexiconService.create(lexicon);
   }
 
+  @Mutation(() => Lexicon)
+  @UseGuards(AuthGuard)
+  async lexiconUpdate(
+  @Args('id') id: string,
+  @Args('name', { nullable: true }) name?: string,
+  @Args('schema', { nullable: true }) schema?: JSONSchema,
+  ): Promise<Lexicon> {
+  const updatedLexicon = await this.lexiconService.update(id, { name, schema });
+  if (!updatedLexicon) {
+    throw new Error(`Lexicon with id ${id} not found`);
```

Comment:
Once you leverage the `LexiconPipe` you can assume the ID is valid. NestJS also comes with error types that correspond to the various HTTP status codes and are returned to the user as such such as

```
BadRequestException(`Lexicon with ID ${lexiconID} does not exist`);
```

Reply:
For this, I have leveraged the LexiconPipe however when I remove the error checker, it provides me with a bunch of errors through the code and this on the return:

Type 'Lexicon | null' is not assignable to type 'Lexicon'.
  Type 'null' is not assignable to type 'Lexicon'.ts(2322)

Reply:
Yea you can use `!` to state explicitly that the value should not be null at that point